File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.cpp
class Solution {
public:
    int countPalindromicSubsequence(string s) {
        int n = s.size();
        int count = 0;
        unordered_set<char> st;
        for (int i = 0; i < n - 2; i++) {
            if (st.count(s[i]) == 0) {
                for (int j = n - 1; j>i+1; j--) {
                    if (s[i] == s[j]) {
                        unordered_set<char>temp;
                        for (int k = i+1; k<j; k++){
                            if (temp.count(s[k]) == 0){
                                temp.insert(s[k]);
                                count++;
                            }
                        }
                        st.insert(s[i]);
                        break;
                    }
                }
            }
        }
        return count;
    }
};

File: 1930-unique-length-3-palindromic-subsequences/1930-unique-length-3-palindromic-subsequences.py
class Solution:
    def countPalindromicSubsequence(self, s: str) -> int:
        count = 0
        first = [-1] * 26
        last = [-1] * 26
        n = len(s)

        for i in range(n):
            idx = ord(s[i]) - ord('a')
            if first[idx] == -1:
                first[idx] = i
            last[idx] = i

        for i in range(26):
            if first[i] != -1 and last[i] - first[i] > 1:
                temp = set()
                for j in range(first[i] + 1, last[i]):
                    temp.add(s[j])
                count += len(temp)

        return count
